### ISSUE TITLE
Add ability to ignore type checking on per file bases

### ISSUE THREAD

**Issue by dahliasalem** [facebook/flow#4464]
If you currently add a file to [ignore] flow will not type check it when running flow --all but it will ALSO completely ignore it's existence which causes a bunch of errors.

Here is my main question:

How difficult would it be to add feature or even an option in .flowconfig to ignore certain files from being typechecked (especially when running with --all) (without ignoring their existence)? It could be a sort of "suppress" comment that is inserted at the beginning of the file to tell flow NOT to typecheck the entire file as opposed to the current suppress comment which works on a per line basis.

This desired behavior is already exhibited when you run flow on files that only contain the "// @flow" annotation. Say file A has "// @flow" annotation but it imports file B which does NOT contain "// @flow" annotation. Currently you are completely free to call on any exports of file B without having flow throwing an error. Can we extend that functionality and make it an actual feature or option in .flowconfig so that it would be helpful when running flow --all?

In other words, currently the only way to tell flow NOT to typecheck is by NOT putting a "// @flow" annotation, can we create a more EXPLICIT way to tell flow not to typecheck a file?

**Comment by jabacchetta** [comment]
Would also be interested in this, specifically in cases when starting a new project with --all

**Comment by SamChou19815** [comment]
`@noflow` exists
